src/main/java/org/jabref/logic/integrity/EditionChecker.java

```diff
@@ -53,6 +53,7 @@ public EditionChecker(BibDatabaseContext bibDatabaseContext, boolean allowIntege
 
         //BibTeX
         if (!bibDatabaseContextEdition.isBiblatexMode()) {
+           if(!isFirstCharDigit(value))
```

Comment:
Please follow our code style: Space after `if`, always use braces.

In case I understood the code correctly, this statement is a fail-fast one. In this case, I would change the code to

```java
if (isFirstCharDigit(value) {
  return Optional.empty();
}
```

Reply:
I tried your solution, but it does skip the method execution